subprojects/kotlin-dsl-provider-plugins/src/main/kotlin/org/gradle/kotlin/dsl/provider/plugins/precompiled/tasks/GeneratePrecompiledScriptPluginAccessors.kt

```diff
@@ -114,8 +133,8 @@ abstract class GeneratePrecompiledScriptPluginAccessors : ClassPathSensitiveCode
 
         val projectPlugins = selectProjectPlugins()
         if (projectPlugins.isNotEmpty()) {
-            withAsynchronousIO(project) {
-                generateTypeSafeAccessorsFor(projectPlugins)
+            asyncIOScopeFactory.newScope().use { scope ->
```

Comment:
💄 it would be nice if we could use `useToRun` here

Reply:
👍 Done